staging/src/k8s.io/cli-runtime/pkg/resource/query_param_verifier.go

```diff
@@ -21,65 +21,67 @@ import (
 	"fmt"
 
 	openapi_v2 "github.com/googleapis/gnostic/openapiv2"
-	yaml "gopkg.in/yaml.v2"
+	"gopkg.in/yaml.v2"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/client-go/discovery"
 	"k8s.io/client-go/dynamic"
 )
 
-func NewDryRunVerifier(dynamicClient dynamic.Interface, openAPIGetter discovery.OpenAPISchemaInterface) *DryRunVerifier {
-	return &DryRunVerifier{
+func NewQueryParamVerifier(dynamicClient dynamic.Interface, openAPIGetter discovery.OpenAPISchemaInterface, queryParam VerifiableQueryParam) *QueryParamVerifier {
+	return &QueryParamVerifier{
 		finder:        NewCRDFinder(CRDFromDynamic(dynamicClient)),
```

Comment:
It looks like that is only used when the GVK cannot be located in the schema. interestingly, it also requires the user to have list permission on CRDs to work. --dry-run=server was much less frequent than validation, which will be true by default. We may need to revisit the fallback to CRDs to make sure we're not modifying the behavior for types that cannot be located server-side

Reply:
> @apelisse NewCRDFinder#cacheCRDs initializes on first use... do we care this work won't be shared across dry-run and field-validator verifiers?

Yes, that's going to cause a double list on CRDs. At this point, I think it's fairly safe to remove the dryRunVerifier though?

> --dry-run=server was much less frequent than validation, which will be true by default. We may need to revisit the fallback to CRDs when we start using this for field validation to make sure we're not modifying the behavior for types that cannot be located server-side

We could catch errors and do client-side validation or no validation at all (probably the former to avoid astonishment) when we can't determine if CRDs support validation?